lib/src/model/class.dart

```diff
@@ -602,4 +602,7 @@ class Class extends Container
       name == o.name &&
       o.library.name == library.name &&
       o.library.package.name == library.package.name;
+
+  @override
+  int get hashCode => quiver.hash3(name, library.name, library.package.name);
```

Comment:
This is not guaranteed to be unique because different private libraries inside a package are allowed to have the same name, confusingly.

Reply:
Aren't hashCodes for equivalent objects(as defined by the `==` operator) supposed to be the same? Taking a look I couldn't really identify why the override of == is actually necessary. I did identify though that the corresponding `hashCode` was removed in https://github.com/dart-lang/dartdoc/pull/1368. 

Perhaps you can provide some context here or a direction to go.

Thanks :D

Reply:
Yes, that's true.  And, two _different_ libraries in the same package are allowed to have the same name.  And two classes, one in each library, are also allowed to have the same name.  These two objects would refer to different classes, but compare equally with this hashCode and therefore confuse other parts of dartdoc.

Reply:
To be explicit here, your hashCode implementation will allow objects that are not equivalent (two classes defined in different libraries with the same name) to compare the same, which is bad.

Reply:
@jcollins-g  That makes sense, thanks. I suppose my question though is, do you know why the pre-existing overridden equal defines equality as that comparison then?

Reply:
Well, er....   ***looks around, embarrassed***

Because it is broken.  That should be deleted.  :-)

Reply:
Sorry about the confusion! I've gone ahead and removed both :)